web_poet/pages.py

```diff
@@ -63,10 +70,98 @@ def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
 
     async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
+
+        partial_item = await self._partial_item()
+        if partial_item:
+            return partial_item
+
         return await item_from_fields(
             self, item_cls=self.item_cls, skip_nonitem_fields=self._skip_nonitem_fields
         )
 
```

Comment:
`_get_select_fields()` is called at least twice on the page but it's value is not being cached.

I think this should be okay since the user might want to re-use the same page instance but with different `SelectFields`:

```
page = SomePage(SelectFields(fields={"*": False, "x": True}))
item1 = page.to_item()

page.select_fields = SelectFields(fields={"*": False, "x": True, "y": True})
item2 = page.to_item()
```